orttraining/orttraining/training_ops/cuda/tensor/resize_grad.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <memory>
+#include <utility>
+
+#include "orttraining/training_ops/cuda/tensor/resize_grad.h"
+#include "orttraining/training_ops/cuda/tensor/resize_grad_impl.h"
+
+namespace onnxruntime::cuda {
+
+#define REGISTER_RESIZEGRAD_KERNEL_TYPED(T)                                \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                           \
+      ResizeGrad,                                                          \
+      kMSDomain,                                                           \
+      1,                                                                   \
+      T,                                                                   \
+      kCudaExecutionProvider,                                              \
+      (*KernelDefBuilder::Create())                                        \
+          .InputMemoryType(OrtMemTypeCPUInput, 2) /* Keep roi on CPU */    \
```

Comment:
how this CPU input restriction get guaranteed during gradient builder? Looks we just take the input of the forward op Resize, who also did not check the input is on CPU?

Reply:
Let's sync offline about this.

Reply:
The forward op marks these inputs as cpu inputs. Additionally, if these inputs are not already on CPU, I believe the allocation planner might move them to cpu if declared this way. I believe this should not result in seg fault as we discussed offline.